### DISCUSSION TITLE
Self hosted runner - Token usage

### DISCUSSION THREAD

**Discussion by carl-mastrangelo** [actions/runner#1921] [Q&A]
Hi, I am trying to builder a docker container with the GitHub runner, but I feel like I'm holding it wrong.   I'd like to be able to make a docker container, almost fully configured, and clone the runner instances on demand (we use GCE VMs).  However, the token I copy paste from the GitHub Runner setup page can only be used on a single runner.  Also, the runner name is setup when the config script is run, rather than when the runner starts up.  

I'm wondering how I can do last minute configuration when the Runner starts (e.g. the name and auth), rather than when the container is built?

**Comment by henrygriffiths** [comment]
Hi @carl-mastrangelo. I'm not quite sure what you are trying to achieve given your description. I know the GitHub token scope is very limited given security implications. For something like this, have you tried creating and using a personal access token from a `real` account (your own, or a bot account you create for this purpose)? I use the token from a bot account extensively for things the GitHub token can't do (such as approving certain pull requests and interacting with other repositories within the organization).
